packages/quantic/force-app/main/default/lwc/quanticRefineModalContent/quanticRefineModalContent.js

```diff
@@ -99,21 +112,34 @@ export default class QuanticRefineModalContent extends LightningElement {
     this.headless = getHeadlessBundle(this.engineId);
     this.searchStatus = this.headless.buildSearchStatus(engine);
     this.breadcrumbManager = this.headless.buildBreadcrumbManager(engine);
+    this.getSortOptionsFromStore();
+
     this.unsubscribeSearchStatus = this.searchStatus.subscribe(() =>
-      this.gatherFacets()
+      this.getFacetDataFromStore()
     );
     this.unsubscribeBreadcrumbManager = this.breadcrumbManager.subscribe(() =>
       this.updateHasActiveFilters()
     );
   };
 
   /**
-   * Gather all facets registered in the Quantic store.
+   * Initializes all facets registered in the Quantic store.
    * @returns {void}
    */
-  gatherFacets() {
+  getFacetDataFromStore() {
     if (!this.hasFacets) {
-      this.data = getAllFacetsFromStore(this.engineId);
+      this.facetData = getAllFacetsFromStore(this.engineId);
+    }
+  }
+
+  /**
+   * Initializes all sort options registered in the Quantic store.
+   * @returns {void}
+   */
+  getSortOptionsFromStore() {
+    this.sortData = getAllSortOptionsFromStore(this.engineId);
+    if (this.sortData?.length) {
+      this.sortOptionsInitialized = true;
     }
   }
```

Comment:
You are using this variable only to determine if you should be rendering the sort section in the refine modal content so it makes total sense for it to be a getter in this case.


```suggestion
  /**
   * Initializes all sort options registered in the Quantic store.
   * @returns {void}
   */
  getSortOptionsFromStore() {
    this.sortData = getAllSortOptionsFromStore(this.engineId);
  }

  get hasSortOptions() {
    return this.sortData?.length > 0;
  }
```